diff -Naur luminance-hdr-2.5.0.orig/CMakeLists.txt luminance-hdr-2.5.0/CMakeLists.txt
--- luminance-hdr-2.5.0.orig/CMakeLists.txt	2017-04-20 22:23:47.159638276 +0200
+++ luminance-hdr-2.5.0/CMakeLists.txt	2017-04-20 22:48:44.207638155 +0200
@@ -282,12 +282,6 @@
     INSTALL(FILES ${CMAKE_SOURCE_DIR}/program-icons/luminance-hdr.png DESTINATION share/icons/hicolor/48x48/apps) # icon
     INSTALL(FILES ${CMAKE_SOURCE_DIR}/luminance-hdr.desktop DESTINATION share/applications) #desktop entry
     INSTALL(FILES ${CMAKE_SOURCE_DIR}/luminance-hdr.appdata.xml DESTINATION share/appdata) #appstream 
-    INSTALL(FILES
-    ${CMAKE_SOURCE_DIR}/AUTHORS
-    ${CMAKE_SOURCE_DIR}/README.md
-    ${CMAKE_SOURCE_DIR}/LICENSE
-    ${CMAKE_SOURCE_DIR}/Changelog
-    DESTINATION share/luminance-hdr) #info files
     INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/help DESTINATION share/luminance-hdr) # help directory
     INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/hdrhtml DESTINATION share/luminance-hdr) # hdrhtml directory
 # Microsoft Windows
