--- ./rtengine/init.cc.orig	2018-08-06 19:02:57.522594783 -0000
+++ ./rtengine/init.cc	2018-08-06 19:04:41.391452565 -0000
@@ -48,20 +48,16 @@
     PerceptualToneCurve::init();
     RawImageSource::init();
 
-#ifdef _OPENMP
-#pragma omp parallel sections
-#endif
-{
-#ifdef _OPENMP
-#pragma omp section
-#endif
-{
     if (s->lensfunDbDirectory.empty() || Glib::path_is_absolute(s->lensfunDbDirectory)) {
         LFDatabase::init(s->lensfunDbDirectory);
     } else {
         LFDatabase::init(Glib::build_filename(baseDir, s->lensfunDbDirectory));
     }
-}
+
+#ifdef _OPENMP
+#pragma omp parallel sections
+#endif
+{
 #ifdef _OPENMP
 #pragma omp section
 #endif
