--- vivovl/eclass/pax-utils.eclass.orig	2018-08-16 13:15:53.202398944 -0000
+++ vivovl/eclass/pax-utils.eclass.new	2018-08-16 13:17:18.587102598 -0000
@@ -53,6 +53,9 @@
 # Either ask on the gentoo-hardened mailing list, or CC/assign
 # hardened@gentoo.org on the bug report.
 pax-mark() {
+	# Unused and fail sometimes in containers
+	return 0
+
 	local f								# loop over paxables
 	local flags							# pax flags
 	local ret=0							# overall return code of this function
