--- python-any-r1.eclass.orig	2018-08-29 18:02:40.784938544 -0000
+++ python-any-r1.eclass.new	2018-08-31 08:42:26.427964538 -0000
@@ -58,6 +58,51 @@
 
 EXPORT_FUNCTIONS pkg_setup
 
+# @ECLASS-VARIABLE: PYTHON_ENABLED_IF_SOURCE
+# @DESCRIPTION:
+# This variable work in only if also ${PYTHON_ENABLE_IF} has a value
+# when "source" python version is found but "dest" is not add 
+# ${PYTHON_ENABLE_IF} to ${PYTHON_COMPAT}
+# Be warned of cache incoherency
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLED_IF_SOURCE=python3_5
+# @CODE
+
+# @ECLASS-VARIABLE: PYTHON_ENABLE_IF
+# @DESCRIPTION:
+# See description for ${PYTHON_ENABLED_IF_SOURCE}
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLE_IF=python3_6
+# @CODE
+
+: ${PYTHON_ENABLED_IF_SOURCE:=python3_6}
+: ${PYTHON_ENABLE_IF:=python3_7}
+: ${PYPY_ENABLED_IF_SOURCE:=python3_5}
+: ${PYPY_ENABLE_IF:=pypy3}
+[[ -n ${PYTHON_ENABLED_IF_SOURCE} && -n ${PYTHON_ENABLE_IF} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLED_IF_SOURCE}} != ${PYTHON_COMPAT[@]} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLE_IF}} == ${PYTHON_COMPAT[@]} ]] \
+&& for _add_python_ver in ${PYTHON_ENABLE_IF} ; do
+	[[ ${PYTHON_COMPAT[@]/${_add_python_ver}} != ${PYTHON_COMPAT[@]} ]] \
+	&& PYTHON_COMPAT+=( ${_add_python_ver} )
+done
+case "${EAPI:-0}" in
+	5|6|7)
+		# EAPI=5 is required
+		[[ -n ${PYPY_ENABLED_IF_SOURCE} && -n ${PYPY_ENABLE_IF} ]] \
+		&& [[ ${PYTHON_COMPAT[@]/${PYPY_ENABLED_IF_SOURCE}} != ${PYTHON_COMPAT[@]} ]] \
+		&& [[ ${PYTHON_COMPAT[@]/${PYPY_ENABLE_IF}} == ${PYTHON_COMPAT[@]} ]] \
+		&& for _add_python_ver in ${PYPY_ENABLE_IF} ; do
+			[[ ${PYTHON_COMPAT[@]/${_add_python_ver}} != ${PYTHON_COMPAT[@]} ]] \
+			&& PYTHON_COMPAT+=( ${_add_python_ver} )
+		done
+		;;
+esac
+
 # @ECLASS-VARIABLE: PYTHON_COMPAT
 # @REQUIRED
 # @DESCRIPTION:
