--- python-any-r1.eclass.orig	2018-03-10 04:19:17.760578919 -0000
+++ python-any-r1.eclass	2018-03-10 13:09:01.260319571 -0000
@@ -57,6 +57,34 @@
 
 EXPORT_FUNCTIONS pkg_setup
 
+# @ECLASS-VARIABLE: PYTHON_ENABLED_IF_SOURCE
+# @DESCRIPTION:
+# This variable work in only if also ${PYTHON_ENABLE_IF} has a value
+# when "source" python version is found but "dest" is not add 
+# ${PYTHON_ENABLE_IF} to ${PYTHON_COMPAT}
+# Be warned of cache incoherency
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLED_IF_SOURCE=python3_5
+# @CODE
+
+# @ECLASS-VARIABLE: PYTHON_ENABLE_IF
+# @DESCRIPTION:
+# See description for ${PYTHON_ENABLED_IF_SOURCE}
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLE_IF=python3_6
+# @CODE
+
+: ${PYTHON_ENABLED_IF_SOURCE:=python3_6}
+: ${PYTHON_ENABLE_IF:=python3_7}
+[[ -n ${PYTHON_ENABLED_IF_SOURCE} && -n ${PYTHON_ENABLE_IF} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLED_IF_SOURCE}} != ${PYTHON_COMPAT[@]} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLE_IF}} == ${PYTHON_COMPAT[@]} ]] \
+&& PYTHON_COMPAT+=( ${PYTHON_ENABLE_IF} )
+
 # @ECLASS-VARIABLE: PYTHON_COMPAT
 # @REQUIRED
 # @DESCRIPTION:
@@ -69,24 +97,6 @@
 # PYTHON_COMPAT=( python{2_5,2_6,2_7} )
 # @CODE
 
-# @ECLASS-VARIABLE: PYTHON_COMPAT_OVERRIDE
-# @INTERNAL
-# @DESCRIPTION:
-# This variable can be used when working with ebuilds to override
-# the in-ebuild PYTHON_COMPAT. It is a string naming the implementation
-# which will be used to build the package. It needs to be specified
-# in the calling environment, and not in ebuilds.
-#
-# It should be noted that in order to preserve metadata immutability,
-# PYTHON_COMPAT_OVERRIDE does not affect dependencies. The value of
-# EPYTHON and eselect-python preferences are ignored. Dependencies need
-# to be satisfied manually.
-#
-# Example:
-# @CODE
-# PYTHON_COMPAT_OVERRIDE='pypy' emerge -1v dev-python/bar
-# @CODE
-
 # @ECLASS-VARIABLE: PYTHON_REQ_USE
 # @DEFAULT_UNSET
 # @DESCRIPTION:
@@ -282,23 +292,6 @@
 python_setup() {
 	debug-print-function ${FUNCNAME} "${@}"
 
-	# support developer override
-	if [[ ${PYTHON_COMPAT_OVERRIDE} ]]; then
-		local impls=( ${PYTHON_COMPAT_OVERRIDE} )
-		[[ ${#impls[@]} -eq 1 ]] || die "PYTHON_COMPAT_OVERRIDE must name exactly one implementation for python-any-r1"
-
-		ewarn "WARNING: PYTHON_COMPAT_OVERRIDE in effect. The following Python"
-		ewarn "implementation will be used:"
-		ewarn
-		ewarn "	${PYTHON_COMPAT_OVERRIDE}"
-		ewarn
-		ewarn "Dependencies won't be satisfied, and EPYTHON/eselect-python will be ignored."
-
-		python_export "${impls[0]}" EPYTHON PYTHON
-		python_wrapper_setup
-		return
-	fi
-
 	# first, try ${EPYTHON}... maybe it's good enough for us.
 	if [[ ${EPYTHON} ]]; then
 		if _python_EPYTHON_supported "${EPYTHON}"; then
