--- python-r1.eclass.orig	2018-08-29 20:02:40.784938544 +0200
+++ python-r1.eclass.new	2018-08-31 23:55:44.611384851 +0200
@@ -54,6 +54,43 @@
 
 fi
 
+# @ECLASS-VARIABLE: PYTHON_ENABLED_IF_SOURCE
+# @DESCRIPTION:
+# This variable work in only if also ${PYTHON_ENABLE_IF} has a value
+# when "source" python version is found but "dest" is not add 
+# ${PYTHON_ENABLE_IF} to ${PYTHON_COMPAT}
+# Be warned of cache incoherency
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLED_IF_SOURCE=python3_5
+# @CODE
+
+# @ECLASS-VARIABLE: PYTHON_ENABLE_IF
+# @DESCRIPTION:
+# See description for ${PYTHON_ENABLED_IF_SOURCE}
+#
+# Example use:
+# @CODE
+# PYTHON_ENABLE_IF=python3_6
+# @CODE
+
+: ${PYTHON_ENABLED_OLD:=python3_6}
+: ${PYTHON_ENABLE_NEW:=python3_7}
+: ${PYPY_ENABLED_OLD:=python3_5}
+: ${PYPY_ENABLE_NEW:=pypy3}
+
+[[ -n ${PYTHON_ENABLED_OLD} && -n ${PYTHON_ENABLE_NEW} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLED_OLD}} != ${PYTHON_COMPAT[@]} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYTHON_ENABLE_NEW}} == ${PYTHON_COMPAT[@]} ]] \
+&& PYTHON_COMPAT+=( ${PYTHON_ENABLE_NEW} )
+
+[[ -n ${PYPY_ENABLED_OLD} && -n ${PYPY_ENABLE_NEW} ]] \
+&& [[ ${EAPI} == [567] ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYPY_ENABLED_OLD}} != ${PYTHON_COMPAT[@]} ]] \
+&& [[ ${PYTHON_COMPAT[@]/${PYPY_ENABLE_NEW}} == ${PYTHON_COMPAT[@]} ]] \
+&& PYTHON_COMPAT+=( ${PYPY_ENABLE_NEW} )
+
 # @ECLASS-VARIABLE: PYTHON_COMPAT
 # @REQUIRED
 # @DESCRIPTION:
