--- etc/service-configuration.xml	2010-10-09 10:25:31.000000000 +0200
+++ etc/service-configuration.xml.syslog	2010-10-09 10:27:10.000000000 +0200
@@ -340,6 +340,7 @@
     <invoke at="status" pass="0" method="status"/>
     <invoke at="stop" pass="0" method="stop"/>
   </service>
+  -->
   <service>
     <name>OpenNMS:Name=Syslogd</name>
     <class-name>org.opennms.netmgt.syslogd.jmx.Syslogd</class-name>
@@ -348,6 +349,7 @@
     <invoke at="status" pass="0" method="status"/>
     <invoke at="stop" pass="0" method="stop"/>
   </service>
+  <!--
   <service>
     <name>OpenNMS:Name=Xmlrpcd</name>
     <class-name>org.opennms.netmgt.xmlrpcd.jmx.Xmlrpcd</class-name>
